vllm/v1/core/scheduler.py

```diff
@@ -247,6 +248,13 @@ def schedule(self) -> "SchedulerOutput":
                         self.encoder_cache_manager.allocate(request, i)
                     encoder_budget = new_encoder_budget
 
+        # Now that requests are scheduled, generate a mask indicating which
+        # request is partial
+        partial_running_reqs = [
+            (req.num_computed_tokens + num_scheduled_tokens[req.request_id] <
+             req.num_tokens) for req in self.running
+        ]
```

Comment:
We should have guaranteed one partial running request right? See `has_partial_request ` in this function.

Reply:
agreed; now only the single index of the one partial request is tracked.